### PR TITLE
[ENH] Add plot_interaction_map utility for AptaTransPipeline

### LINKED ISSUE #283: [ENH] Create a plot_interaction_map utility function for AptaTransPipeline
**Is your feature request related to a problem? Please describe.**
Yes. Currently, in pyaptamer/aptatrans/_pipeline.py, the get_interaction_map() method returns a raw PyTorch interaction map tensor. As noted by an explicit TODO on line 156, those using the library currently have no built-in way to easily visualize this output map

**Describe the solution you'd like**
I'd like to provide a ready-to-use utility function (e.g., plot_interaction_map()) that takes the raw tensor output of get_interaction_map(), along with the candidate and target sequences as labels. This function would use matplotlib or seaborn to render a beautiful 2D heatmap showing the attention/interaction intensity between the aptamer nucleotides and protein residues.

Ideally, this function could live either inside _pipeline.py or within a new/existing visualization utilities module, ensuring the axes are clearly labeled (X-axis: Protein residues, Y-axis: Aptamer nucleotides).

**Describe alternatives you've considered**
An alternative is leaving the visualization entirely up to the user. However, since the AptaTrans pipeline intrinsically relies on interaction mapping for interpretability, users would end up constantly rewriting the same boilerplate plotting code. Offering it as an out-of-the-box utility significantly improves the library's utility.

**Additional context**
Relevant File: pyaptamer/aptatrans/_pipeline.py (around line 156).
The plotting logic and styling should roughly mimic the heatmap visualizations shown in the original AptaTrans publication to ensure consistency.

### PR DESCRIPTION
**Pull request by tarun-227**
#### Reference Issues/PRs

Fixes #283.

#### What does this implement/fix? Explain your changes.

Adds a `plot_interaction_map()` utility function that renders the raw interaction map tensor from `AptaTransPipeline.get_interaction_map()` as a 2D heatmap using matplotlib.

Usage:
```python
from pyaptamer.utils import plot_interaction_map

imap = pipeline.get_interaction_map(candidate="ACGUA", target="DHRNENAI")
ax = plot_interaction_map(imap, candidate="ACGUA", target="DHRNENAI")
```

Features:
- Accepts the raw output tensor of shape `(batch, 1, seq_len_apta, seq_len_prot)` or pre-squeezed `(H, W)`
- Handles both numpy arrays and torch tensors
- Labels axes with individual nucleotides/residues from the candidate and target sequences
- Customizable colormap, figure size, and axes (pass an existing `ax` for subplot layouts)
- Extra kwargs forwarded to `ax.imshow()` for full control

Files added/changed:
- `pyaptamer/utils/_plotting.py` — `plot_interaction_map()` and helper `_squeeze_interaction_map()`
- `pyaptamer/utils/tests/test_plotting.py` — 8 tests
- `pyaptamer/utils/__init__.py` — registered in imports and `__all__`

#### What should a reviewer concentrate their feedback on?

- Whether `matplotlib` should be added as an optional dependency in `pyproject.toml` (currently imported lazily inside the function to avoid making it a hard requirement)
- Whether the function should live here in `utils/` or in a new `pyaptamer/plotting/` subpackage
- Colormap and styling preferences — currently defaults to `viridis` with a colorbar

#### Did you add any tests for the change?

Yes, 8 tests in `pyaptamer/utils/tests/test_plotting.py`:
- Standard 4D and 2D input shapes
- Sequence labels on axes (candidate and target)
- Custom axes and colormap
- Torch tensor input handling
- Invalid shape raises `ValueError`
- Labels truncated when sequences are longer than the map dimensions

#### Any other comments?

`matplotlib` is imported lazily inside the function so it doesn't become a hard dependency for users who don't need plotting.

#### PR checklist

- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG].
- [x] Added/modified tests
- [x] Used pre-commit hooks when committing to ensure that code is compliant with hooks.

### CODE REVIEW

**Comment by tarun-227** [comment]
Hey @satvshr and @NennoMP, could you please take a look at this PR when you get a chance? Would appreciate any feedback or review. Thanks!

**Comment by NennoMP** [comment]
Same issues of #293, I copy-paste my comment from there.

(i) The TODO comment explicitly mentioned that an idea was to leave the untouched interaction map as is, rather than plot it. Before starting a PR, it would have made sense to at least have a discussion with maintainers within the corresponding issue.
(ii) In my opinion, if we really want to add a plotting functionality, it should be within pyaptamer.aptatrans. The interaction map is unique so AptaTrans, so I would not put a method used by just one algorithm in utils.
(iii) Most importantly, plotting should be implemented as a refactored version of the original authors' code (see method get_interaction_map at [link](https://github.com/PNUMLB/AptaTrans/blob/master/aptatrans_pipeline.py)).

Related to point (ii), I also don't see the plotting method being called and/or made available within the AptaTrans pipeline. Basically, the user is not able to use it via the pipeline right now, am I missing something?

**Comment by tarun-227** [comment]
Thanks for the detailed feedback @NennoMP! I've reworked the PR to address all three points:

**(i) Discussed before starting** — noted for future PRs, apologies for jumping ahead.

**(ii) Moved to `pyaptamer/aptatrans/`** — `_plotting.py` now lives inside `aptatrans/`, not `utils/`. The `utils/` files and tests have been removed.

**(iii) Refactored from original authors' code** — `plot_interaction_map()` is now a method on `AptaTransPipeline`, mirroring the logic in the original [`get_interactionmap()`](https://github.com/PNUMLB/AptaTrans/blob/master/aptatrans_pipeline.py): tokenizes both sequences, trims the raw map to non-padded token lengths, decodes 3-mer token labels, applies softmax per `view` parameter (`"apta"`, `"prot"`, or `None` for combined), and highlights the top-k interacting tokens on the axes.

**(iv) Accessible via the pipeline** — users can now call it directly:
```python
ax = pipeline.plot_interaction_map(candidate="ACGUA", target="DHRNENAI", view="apta", top_k=5)
```

5 new tests added, all 27 passing (1 skipped for CUDA). Would appreciate another look!
